### DISCUSSION TITLE
List of benchmarks / toy problems for possible inclusion

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1576] [Ideas/Science]
*Update by Michael*: I've repurposed this ticket to collect various toy problems / benchmarks we might like to add at some point. Whenever someone decides a certain benchmark is necessary - and that they're going to implement it - we can make a separate ticket for that!

- [Hass et al. (2018 pre-print)](https://www.biorxiv.org/content/early/2018/08/30/404590) - Reaction-diffusion problems, Optimisation focus
  - 20 biochemical reaction networks, all taken from literature & with real data.
  - Note in discussion that log transform is good because convexity changes
- [Villaverde et al. "BioPredyn" (2015)](https://bmcsystbiol.biomedcentral.com/articles/10.1186/s12918-015-0144-4) - All reaction-diffusion problems, Optimisation focus
  - Problem B1: genome-wide kinetic model of S. cerevisiae
  - Problem B2: dynamic model of the Central Carbon Metabolism of E. coli
  - Problem B3: enzymatic and transcriptional regulation of the Central Carbon Metabolism of E. coli
  - Problem B4: metabolic model of Chinese Hamster Ovary (CHO) cells
  - Problem B5: signal transduction logic model
  - Problem B6: the gap gene network of the vinegar fly, Drosophila melanogaster 
- [Ballnus et al (2017).](https://bmcsystbiol.biomedcentral.com/articles/10.1186/s12918-017-0433-1) - All reaction-diffusion problems, MCMC focus
  - Very nice plots of the posterior (or a representative cut) for each problem
  - M1 mRNA transfection
  - M2 Bistable switch
  - M3 Saturated growth
  - M4 Biochemical system with Hopf bifurcation
  - M5 Driven Van der Pol Oscillator
  - M6 Lorenz Attractor
- [Feroz & Skilling (2013)](https://arxiv.org/pdf/1312.5638.pdf) - Theoretical focus
  - Rastrigin function,
  - Himmelblau's function,
  - multidimensional eggbox,
  - Bayesian object detection case study.
- [Wikipedia's list of optimisation test functions](https://en.wikipedia.org/wiki/Test_functions_for_optimization)

**Comment by MichaelClerx** [comment]
One the one hand, I'm all for including these.
But on the other hand, every new toy problem means another row/column in our performance testing matrix, plus a lot more code to document and maintain. We have to be on the look-out for bloat!

**Comment by ben18785** [comment]
Nice [list](https://en.wikipedia.org/wiki/Test_functions_for_optimization) of functions commonly used in mathematical optimisation.

Agreed @MichaelClerx, just listing stuff which we can decide about later.

**Comment by ben18785** [comment]
For MCMC, not sure that Himmelblau will be that useful although may be for optimisation.

**Comment by MichaelClerx** [comment]
@chonlei found that Villaverde et al. have done another optimisation benchmark paper https://academic.oup.com/bioinformatics/article/35/5/830/5078478
